### PR TITLE
Support for sensitive environment variables

### PR DESCRIPTION
**Pull request by teluq-pbrideau**
#### Pull Request (PR) description
The `environment` option does not support sensitive value.  I would like to put `http_proxy=http://user:password@example.com:3128` in an environment variable, but the content is displayed in cleartext in the log.

This PR allow to define an `environment` content as sensitive as followed:
```puppet
cron::job { 'example' :
  environment => [
    Sensitive('http_proxy=http://user:password@example.com:3128'),
    'PATH=/usr/bin',
  ],
  command     => 'sleep 10',
}
```
same for `cron::job::multiple`

There was validation in the `erb` template about the `environment` variables, and funky `.join(\n).split(\n)` but I think it is no longer necessary.  The content is validated by puppet directly on the call to `cron::job`.  Is this behavior a remnant of old puppet when there was no types?

I also removed the `environment` variable in the `cron::job::multiple::jobs` array, as it was not defined in the old `erb`, and kept the same behavior for the `epp`.

Feel free to raise concern if you think my assumptions were wrong

### CODE REVIEW

File: spec/defines/job_spec.rb [hunk 1]
```diff
@@ -135,6 +135,21 @@
     end
   end
 
+  context 'job with sensitive environment' do
+    let(:params) do
+      {
+        command: '/bin/true',
+        environment: [sensitive('KEY=value')]
+      }
+      let(:cron_timestamp) { get_timestamp(params) }
```
Comment (bastelfreak) [inline spec/defines/job_spec.rb:144]: the let needs to be outside the let params block.

Comment (teluq-pbrideau) [reply spec/defines/job_spec.rb]: well, a simple `bundle update` allowed me to run the tests on my side

**Comment by teluq-pbrideau** [comment]
@bastelfreak Would it be something wanted if I work further on this PR?  Or should I abandon this completely?  I don’t want to work on something that will be discarded…

**Comment by bastelfreak** [comment]
thanks a lot for updating it! I will do a minor release in https://github.com/voxpupuli/puppet-cron/pull/140 and afterward merge your PR.

**Pull request merged**